### ISSUE TITLE
Doesn't seem to work with Python 2.7

### ISSUE THREAD

**Issue by retnuh** [EelcoHoogendoorn/Numpy_arraysetops_EP#1]
Hi there, I tried to install this in a docker image that has Python 2.7 installed.  I got the following error:

``` bash
# pip install numpy-indexed
Collecting numpy-indexed
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading numpy-indexed-0.3.4.zip
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ClI9nN/numpy-indexed/setup.py", line 4, in <module>
        import pkg_conf
      File "/tmp/pip-build-ClI9nN/numpy-indexed/pkg_conf.py", line 2, in <module>
        import yaml
    ImportError: No module named yaml

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-ClI9nN/numpy-indexed/
```

**Comment by TaylorMutch** [comment]
Looks like you need a yaml installation.

`pip install pyyaml`

**Comment by ivoflipse** [comment]
It probably wouldn't help in this case to add a requires section to setup.py, since the error get's throw before `setup()` is executed. The conda package doesn't have this problem, since its dependencies are defined outside setup.py.

See: https://github.com/EelcoHoogendoorn/Numpy_arraysetops_EP/blob/master/conda-recipe/meta.yaml

You'll probably also need `future`, for Python 2-3 compatibility

**Comment by EelcoHoogendoorn** [comment]
I thought I had fixed this by adding install_requires=['pyyaml', 'future'], to setup.py; but as usual when I try anything with setuptools/distutils, it does not work. Does anyone have a clue what is missing there?

**Comment by Korijn** [comment]
Not all versions of pip correctly check `install_requires`, which version was used by OP?

**Comment by retnuh** [comment]
Pip 8.1.2 - here's the output from the original install attempt.  I had upgraded pip in the previous command (shown).  The first two lines were printed out after installing a different package.

```{bash}
You are using pip version 8.0.2, however version 8.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
root@06ba519fed36:/notebooks# pip install --upgrade pip
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:315: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:120: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Collecting pip
  Downloading pip-8.1.2-py2.py3-none-any.whl (1.2MB)
    100% |################################| 1.2MB 453kB/s 
Installing collected packages: pip
  Found existing installation: pip 8.0.2
    Uninstalling pip-8.0.2:
      Successfully uninstalled pip-8.0.2
Successfully installed pip-8.1.2
root@06ba519fed36:/notebooks# pip install numpy-indexed
Collecting numpy-indexed
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading numpy-indexed-0.3.4.zip
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ClI9nN/numpy-indexed/setup.py", line 4, in <module>
        import pkg_conf
      File "/tmp/pip-build-ClI9nN/numpy-indexed/pkg_conf.py", line 2, in <module>
        import yaml
    ImportError: No module named yaml
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-ClI9nN/numpy-indexed/
```

**Comment by EelcoHoogendoorn** [comment]
Ah yes; setup_requires covers usage of these packages inside the setup call; the problem is I am relying on these packages in the making of the setup call... ill try to figure out what the recommended way of dealing with this is.

**Comment by ivoflipse** [comment]
Probably not relying on 3rd party packages to execute setup.py

**Comment by Korijn** [comment]
This has been fixed by #4 so the issue can be closed @ivoflipse
